include/astshim/ChebyMap.h

```diff
@@ -0,0 +1,321 @@
+/*
+ * LSST Data Management System
+ * Copyright 2017 AURA/LSST.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <https://www.lsstcorp.org/LegalNotices/>.
+ */
+#ifndef ASTSHIM_CHEBYMAP_H
+#define ASTSHIM_CHEBYMAP_H
+
+#include <algorithm>
+#include <memory>
+#include <vector>
+
+#include "astshim/base.h"
+#include "astshim/Mapping.h"
+
+namespace ast {
+
+class ChebyDomain {
+public:
+    /**
+    Construct a ChebyDomain
+
+    @param[in] lbnd  Lower bounds of domain (one element per axis)
+    @param[in] ubnd  Upper bounds of domain (one element per axis)
+    */
+    ChebyDomain(std::vector<double> const &lbnd, std::vector<double> const &ubnd) : lbnd(lbnd), ubnd(ubnd) {}
+    std::vector<double> lbnd;  ///< lower bound of domain (one element per axis)
+    std::vector<double> ubnd;  ///< upper bound of domain (one element per axis)
+};
+
+/**
+A ChebyMap is a form of Mapping which performs a Chebyshev polynomial
+transformation.  Each output coordinate is a linear combination of
+Chebyshev polynomials of the first kind, of order zero up to a
+specified maximum order, evaluated at the input coordinates. The
+coefficients to be used in the linear combination are specified
+separately for each output coordinate.
+
+For a 1-dimensional ChebyMap, the forward transformation is defined
+as follows:
+
+   f(x) = c0.T0(x') + c1.T1(x') + c2.T2(x') + ...
```

Comment:
Good point. I just copied this from AST docs, but the dots are confusing. I'll remove them and hope that suffices.